### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by zhengyangl**
Put isCmp() and isShift() in Inst.cpp
Do not reinitialize Vars in different guesses.

### CODE REVIEW

File: lib/Infer/ExhaustiveSynthesis.cpp [hunk 1]
```diff
@@ -445,6 +434,10 @@ ExhaustiveSynthesis::synthesize(SMTLIBSolver *SMTSolver,
   // find the valid one
   int Unsat = 0;
   int GuessIndex = -1;
+
+  std::vector<Inst *> Vars;
+  findVars(LHS, Vars);
```
Comment (regehr) [inline lib/Infer/ExhaustiveSynthesis.cpp]: what's the reason for moving this code earlier?

**Comment by zhengyangl** [comment]
This code is inside the again: label and evaluated on each constant synthesis

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

File: lib/Infer/ExhaustiveSynthesis.cpp (hunk 1 above)
Comment (zhengyangl) [reply lib/Infer/ExhaustiveSynthesis.cpp]: This code is inside the again: label and evaluated on each constant guess try. I am only moving this before the again: label

Comment (regehr) [reply lib/Infer/ExhaustiveSynthesis.cpp]: aha, great catch, thanks!

**Comment by regehr** [comment]
LGTM

**Pull request merged**